### PR TITLE
[MU4] Fix MSVC/MinGW compiler warnings, fix MinGW compiler error

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
* Fix MSVC compiler warnings:
    * reg. declaration hides class member (C4458)
    * reg. macro redefinition (C4005)
    * reg. conversion from 'size_t' to 'int', possible loss of data (C4267)
    * reg. unreachable code (C4702)

* Fix MinGW compiler error:
    * reg. a header file not found
    Still doesn't link though, due to "undefined reference to `PKEY_Device_FriendlyName'"

* Fix MinGW compiler warning:
    * reg. comparison of integer expressions of different signedness (-Wsign-compare)

* Fix MOC compiler warning:
    * reg. name 'label' is already in use

### CODE REVIEW

**Comment by vpereverzev** [comment]
I'd encourage to not touch 3rd-party sources here

File: src/appshell/view/devtools/settingslistmodel.cpp [hunk 1]
```diff
@@ -103,6 +103,7 @@ QString SettingListModel::typeToString(Val::Type t) const
     case Val::Type::Color:     return "Color";
     case Val::Type::List:      return "List";
     case Val::Type::Map:       return "Map";
+    default: break;
```
Comment (Jojo-Schmitz) [inline src/appshell/view/devtools/settingslistmodel.cpp:106]: Will conflict with #12239

File: thirdparty/google_crashpad_client/third_party/mini_chromium/mini_chromium/base/compiler_specific.h [hunk 2]
```diff
@@ -79,7 +79,9 @@
 #elif defined(__clang__)
 #define FALLTHROUGH [[clang::fallthrough]]
 #else
+#ifndef FALLTHROUGH
 #define FALLTHROUGH
 #endif
```
Comment (cbjeukendrup) [inline thirdparty/google_crashpad_client/third_party/mini_chromium/mini_chromium/base/compiler_specific.h:84]: I think we should rather rename our own macro to MU_FALLTHROUGH (in log.h)

**Comment by cbjeukendrup** [comment]
Probably we'll have to disable the warning in flac by adding another `target_no_warning` in thirdparty/flac/CMakeLists.txt.

File: thirdparty/google_crashpad_client/third_party/mini_chromium/mini_chromium/base/compiler_specific.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply thirdparty/google_crashpad_client/third_party/mini_chromium/mini_chromium/base/compiler_specific.h:84]: Ah, yes, i didn't even check where the conflict is

**Comment by Jojo-Schmitz** [comment]
I'll go that a try

File: thirdparty/google_crashpad_client/third_party/mini_chromium/mini_chromium/base/compiler_specific.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply thirdparty/google_crashpad_client/third_party/mini_chromium/mini_chromium/base/compiler_specific.h]: Done. Actually that marco isn't really needed as `// FALLTHROUGH` comment works just fine cross-platform and is used in several places already, whereas that `FALLTHROUGH()` marco is use in only 2 places (and one file)

**Pull request merged**